nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/main/java/org/apache/nifi/avro/AvroTypeUtil.java

```diff
@@ -691,6 +697,10 @@ private static boolean isCompatibleDataType(final Object value, final DataType d
                     return true;
                 }
                 break;
+            case MAP:
+                if (value instanceof MapRecord) {
```

Comment:
We can probably use the more generic Record instead of MapRecord. We should probably also consider if value instanceof java.util.Map

Reply:
OK, i'll have a dive on this